### PR TITLE
HTS-914: DWP health check

### PR DESCRIPTION
**Pull request by AndyHWChung**

### CODE REVIEW

File: app/uk.gov.hmrc.helptosaveproxy/health/DWPConnectionHealthCheck.scala [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+    implicit val hc: HeaderCarrier = HeaderCarrier()
+
+    override def performTest(): Future[HealthCheckResult] = {
+      val timer = metrics.dwpHealthCheckTimer.time()
+
+      dwpConnector.healthCheck().value
+        .map { result ⇒
+          val time = timer.stop()
+          result.fold[HealthCheckResult](e ⇒ HealthCheckResult.Failure(e, time), _ ⇒ HealthCheckResult.Success("health check returned 200 OK", time))
```
Comment (sureshhmrc) [inline app/uk.gov.hmrc.helptosaveproxy/health/DWPConnectionHealthCheck.scala]: prefix messages with `dwp` ?

File: app/uk.gov.hmrc.helptosaveproxy/health/DWPConnectionHealthCheck.scala [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+      lifecycle),
+      s"health-check-$name-lock"
+    )
+
+  // start the health check only if it is enabled
+  if (enabled) {
+    logger.info(s"HealthCheck $name enabled")
+    val _ = lockedHealthCheck
+  } else {
+    logger.info(s"HealthCheck $name not enabled")
```
Comment (sureshhmrc) [inline app/uk.gov.hmrc.helptosaveproxy/health/DWPConnectionHealthCheck.scala]: prefix messages with dwp ?

Comment (AndyHWChung) [reply app/uk.gov.hmrc.helptosaveproxy/health/DWPConnectionHealthCheck.scala]: `$name` in this case expands to `dwp-connection`

File: app/uk.gov.hmrc.helptosaveproxy/health/DWPConnectionHealthCheck.scala (hunk 1 above)
Comment (AndyHWChung) [reply app/uk.gov.hmrc.helptosaveproxy/health/DWPConnectionHealthCheck.scala]: sure

**Pull request merged**